src/main/java/frc/robot/RobotContainer.java

```diff
@@ -32,73 +32,120 @@
-
-  /**
-   * The container for the robot. Contains subsystems, OI devices, and commands.
-   */
-  public RobotContainer() {
-    switch (Constants.currentMode) {
-      // Real robot, instantiate hardware IO implementations
-      case REAL:
-        drive = new Drive(new DriveIOSparkMax());
-        flywheel = new Flywheel(new FlywheelIOSparkMax());
-        // drive = new Drive(new DriveIOFalcon500());
-        // flywheel = new Flywheel(new FlywheelIOFalcon500());
-        break;
-
-      // Sim robot, instantiate physics sim IO implementations
-      case SIM:
-        drive = new Drive(new DriveIOSim());
-        flywheel = new Flywheel(new FlywheelIOSim());
-        break;
-
-      // Replayed robot, disable IO implementations
-      default:
-        drive = new Drive(new DriveIO() {
-        });
-        flywheel = new Flywheel(new FlywheelIO() {
-        });
-        break;
+    // Subsystems
+    private final Drive drive;
+//    private final Flywheel flywheel;
+
+    // Controller
+    private final CommandXboxController controller = new CommandXboxController(0);
+
+    // Dashboard inputs
+    private final LoggedDashboardChooser<Command> autoChooser = new LoggedDashboardChooser<>("Auto Choices");
+
+    // Battery IO
+    BatteryIO battery;
+    BatteryIOInputsAutoLogged batteryInputs = new BatteryIOInputsAutoLogged();
+    /**
+     * The container for the robot. Contains subsystems, OI devices, and commands.
+     */
+    public RobotContainer() throws IOException {
+        var FRDriveMotorId = 10;
+        var BLDriveMotorId = 11;
+        var FLDriveMotorId = 12;
+        var BRDriveMotorId = 13;
+
+        var FRTurnMotorId = 14;
+        var BLTurnMotorId = 15;
+        var FLTurnMotorId = 16;
+        var BRTurnMotorId = 17;
+
+        var FRTurnEncoderId = 6;
+        var BLTurnEncoderId = 7;
+        var FLTurnEncoderId = 8;
+        var BRTurnEncoderId = 9;
+
+        var gyroID = 20;
+        switch (Constants.currentMode) {
+            // Real robot, instantiate hardware IO implementations
+            case REAL:
+                battery = new BatteryIORio();
+                drive = new Drive(
+                        new ModuleIOFalcon500(FRTurnMotorId, FRDriveMotorId, false, false, FRTurnEncoderId),
+                        new ModuleIOFalcon500(BLTurnMotorId, BLDriveMotorId, false, false, BLTurnEncoderId),
+                        new ModuleIOFalcon500(FLTurnMotorId, FLDriveMotorId, false, false, FLTurnEncoderId),
+                        new ModuleIOFalcon500(BRTurnMotorId, BRDriveMotorId, false, false, BRTurnEncoderId),
+                        new GyroIOReal(gyroID, "rio"),
+                        new VisionIOPhoton());
+//                flywheel = new Flywheel(new FlywheelIOSparkMax());
+                break;
+
+            // Sim robot, instantiate physics sim IO implementations
+            case SIM:
```

Comment:
Pretty sure this is right but just making sure this shouldn't be real I think you got it right here though